app/src/main/java/io/homeassistant/companion/android/sensors/SensorsSettingsFragment.kt

```diff
@@ -116,7 +117,12 @@ class SensorsSettingsFragment : PreferenceFragmentCompat() {
                     }
                 }
                 if (!permArray.isNullOrEmpty())
-                    requestPermissions(permArray, 0)
+                    if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.Q) {
+                        requestPermissions(permArray.toList().minus(Manifest.permission.ACCESS_BACKGROUND_LOCATION).toTypedArray(), 0)
```

Comment:
It looks like this should work but if a user doesn't enable during onboarding and comes to this screen to enable all we still hit the error it seems?

```
2020-10-16 12:31:07.485 17981-17981/? E/GrantPermissionsActivity: Apps targeting 30 must have foreground permission before requesting background and must request background on its own.
2020-10-16 12:31:07.486 17981-17981/? V/GrantPermissionsActivity: Permission grant result requestId=-3303380240007338652 callingUid=10456 callingPackage=io.homeassistant.companion.android.debug permission=android.permission.BIND_NOTIFICATION_LISTENER_SERVICE isImplicit=false result=1
2020-10-16 12:31:07.492 1586-5074/? W/ActivityTaskManager: Duplicate finish request for r=ActivityRecord{7b3811c u0 com.google.android.permissioncontroller/com.android.permissioncontroller.permission.ui.GrantPermissionsActivity t30465 f}}
2020-10-16 12:31:07.507 1586-1990/? I/ActivityTaskManager: START u0 {act=android.content.pm.action.REQUEST_PERMISSIONS pkg=com.google.android.permissioncontroller cmp=com.google.android.permissioncontroller/com.android.permissioncontroller.permission.ui.GrantPermissionsActivity (has extras)} from uid 10456
2020-10-16 12:31:07.516 1586-1610/? D/EventSequenceValidator: IntentStarted during UNKNOWN. Intent { act=android.content.pm.action.REQUEST_PERMISSIONS pkg=com.google.android.permissioncontroller (has extras) }
    java.lang.Throwable: EventSequenceValidator#getStackTrace
        at com.google.android.startop.iorap.EventSequenceValidator.logWarningWithStackTrace(EventSequenceValidator.java:260)
        at com.google.android.startop.iorap.EventSequenceValidator.onIntentStarted(EventSequenceValidator.java:106)
        at com.android.server.wm.LaunchObserverRegistryImpl.handleOnIntentStarted(LaunchObserverRegistryImpl.java:139)
        at com.android.server.wm.LaunchObserverRegistryImpl.lambda$veRn_GhgLZLlOHOJ0ZYT6KcfYqo(Unknown Source:0)
        at com.android.server.wm.-$$Lambda$LaunchObserverRegistryImpl$veRn_GhgLZLlOHOJ0ZYT6KcfYqo.accept(Unknown Source:10)
        at com.android.internal.util.function.pooled.PooledLambdaImpl.doInvoke(PooledLambdaImpl.java:292)
        at com.android.internal.util.function.pooled.PooledLambdaImpl.invoke(PooledLambdaImpl.java:201)
        at com.android.internal.util.function.pooled.OmniFunction.run(OmniFunction.java:97)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:223)
        at android.os.HandlerThread.run(HandlerThread.java:67)
        at com.android.server.ServiceThread.run(ServiceThread.java:44)
2020-10-16 12:31:07.565 17981-17981/? W/GrantPermissionsActivity: Cannot grant android.permission.ACCESS_BACKGROUND_LOCATION as the matching foreground permission is not already granted.
2020-10-16 12:31:07.565 17981-17981/? V/GrantPermissionsActivity: Permission grant result requestId=1625869301739567180 callingUid=10456 callingPackage=io.homeassistant.companion.android.debug permission=android.permission.ACCESS_BACKGROUND_LOCATION isImplicit=false result=1
```

Reply:
It does work as expected on SensorDetailFragment but for some reason enable all isn't liking it

Reply:
hmm is it because there is more than 1 instance of the permission because we loop through collecting all permissions?

Reply:
That very well could be, I'll take a look the next chance I get.

Reply:
It looks like the loop was causing issues here, I think :) 

At line 108 add:

```
                        if (Manifest.permission.ACCESS_BACKGROUND_LOCATION in permArray)
                            if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.Q) {
                                permArray = permArray.toList().minus(Manifest.permission.ACCESS_BACKGROUND_LOCATION)
                                    .toTypedArray()
                            }
```

😃

Reply:
Fixed